tests/compilation_cache_test.py

```diff
@@ -0,0 +1,142 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+import hashlib
+from jax.experimental.compilation_cache import compilation_cache as cc
+import jax
+import jax.test_util as jtu
+import numpy as np
+import unittest
+
+class CompilationCacheTest(jtu.JaxTestCase):
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_cache_key_type(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    self.assertEqual(type(cc.get_cache_key(computation, compile_options)), str)
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_compile_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.filled_compile_options()
+    filled_hash1 = self.get_hashed_value(cc._hash_compile_options, compile_options_filled)
+    filled_hash2 = self.get_hashed_value(cc._hash_compile_options, compile_options_filled)
+    filled_hash3 = self.get_hashed_value(cc._hash_compile_options, compile_options_not_filled)
+    self.assertEqual(filled_hash1, filled_hash2)
+    self.assertNotEqual(filled_hash1, filled_hash3)
+
+  @unittest.skipIf(jax.lib.version < (0, 1, 68), "fails with earlier jaxlibs")
+  def test_executable_build_options(self):
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    compile_options_filled = self.filled_compile_options()
+    filled_hash1 = self.get_hashed_value(cc._hash_executable_build_options,
+                   compile_options_filled.executable_build_options)
+    filled_hash2 = self.get_hashed_value(cc._hash_executable_build_options,
+                   compile_options_filled.executable_build_options)
+    filled_hash3 = self.get_hashed_value(cc._hash_executable_build_options,
+                   compile_options_not_filled.executable_build_options)
+    self.assertEqual(filled_hash1, filled_hash2)
+    self.assertNotEqual(filled_hash1, filled_hash3)
+
+  def test_debug_options(self):
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    filled_hash1 = self.get_hashed_value(cc._hash_debug_options,
+                compile_options.executable_build_options.debug_options)
+    filled_hash2 = self.get_hashed_value(cc._hash_debug_options,
+                compile_options.executable_build_options.debug_options)
+    self.assertEqual(filled_hash1, filled_hash2)
```

Comment:
Given that you don't actually test different DebugOptions anywhere else, I think you should manually create a different DebugOptions object here and test that it gets a different hash. You can do so by modifying the fields of an existing DebugOptions.